File: codes/two-sum.py
from typing import List


class Solution:
    @staticmethod
    def twoSum(nums: List[int], target: int) -> List[int]:
        iterated = {}

        for i, num in enumerate(nums):
            diff = target - num
            if diff in iterated:
                return [iterated[diff], i]

            iterated[num] = i


print(Solution.twoSum([2, 7, 11, 15], 9))  # [0, 1]

File: codes/arrays_and_hashing/two-sum.rb
# frozen_string_literal: true

# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  iterated = {}

  nums.each_with_index do |num, i|
    diff = target - num
    return [iterated[diff], i] if iterated[diff]

    iterated[num] = i
  end
end

p two_sum [2, 7, 11, 15], 9 # [0, 1]
